src/Common/OptimizedRegularExpression.h

```diff
@@ -95,6 +95,15 @@ class OptimizedRegularExpressionImpl
         out_required_substring_is_prefix = required_substring_is_prefix;
     }
 
+    /// analyze function will extract the longest string literal or multiple alternative string literals from regexp for pre-checking if
```

Comment:
If I read the test (gtest_optimize_re.cpp) right, then `required_substring` is not required to be a prefix of the regexp but each `alternative` is. Perhaps worth to mention.